snowdrop-execution/src/Snowdrop/Execution/DbActions/AVLp/Actions.hs

```diff
@@ -93,69 +95,116 @@ avlServerDbActions = fmap mkActions . newTVarIO
     mkActions var = (\recForProof ->
                         DbModifyActions
                           (mkAccessActions var recForProof)
-                          (reThrowAVLEx @k . apply var),
+                          (apply var),
                         retrieveHash var)
     mkAccessActions var recForProof =
         DbAccessActions
           -- adding keys to amsRequested
-          (\cA req -> liftIO $ reThrowAVLEx @k $ query cA req =<<
-            atomically (retrieveAMS var recForProof $ AMSKeys req ))
-          (\cA cs -> liftIO $ reThrowAVLEx @k $ modAccum cA cs =<< atomically (readTVar var))
+          (\cA req ->
+            liftIO (atomically (retrieveAMS var recForProof $ rmap (AMSKeys . unHSetEl) req))
+                >>= \ctx -> query ctx cA req
+          )
+          (\cA cs ->
+             liftIO (atomically (readTVar var)) >>= \ctx -> modAccum ctx cA cs)
           -- setting amsRequested to AMSWholeTree as iteration with
           -- current implementation requires whole tree traversal
-          (\cA p b f -> liftIO $ reThrowAVLEx @k $ iter cA p b f =<<
-            atomically (retrieveAMS var recForProof AMSWholeTree ))
-
+          (\cA ->
+             liftIO (atomically (retrieveAMS var recForProof allWholeTree))
```

Comment:
Why all whole tree? In fact, only a particular component shall be set to `AMSWholeTree`